src/operator/tensor/init_op.h

```diff
@@ -320,11 +328,13 @@ inline bool RangeShape(const nnvm::NodeAttrs& attrs,
       << "Range does not support (start, stop, step)= "
       << "(" << param.start << "," << param.stop.value() << "," << param.step << ")";
   }
-  SHAPE_ASSIGN_CHECK(*out_attrs, 0,
-                     mshadow::Shape1(mshadow::expr::RangeOutSize(param.start,
-                                                                 param.stop.value(),
-                                                                 param.step,
-                                                                 param.repeat)));
+  MSHADOW_TYPE_SWITCH(param.dtype, DType, {
+    DType start = static_cast<DType>(param.start);
```

Comment:
Should the cast happen here? For example if DType is int8 but range is 0 - 512, the result should be 0, 1, ..., 255, 0, ..., 255